region/tourian/main/Metroid Room 2.json

```diff
@@ -343,6 +343,60 @@
         {"metroidFrames": 80}
       ]
     },
+    {
+      "link": [2, 1],
+      "name": "Very Deep Stuck X-Ray Climb With Ice",
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "direct",
+          "morphed": false
+        }
+      },
+      "requires": [
+        "Ice",
+        "canXRayClimb"
+      ],
+      "bypassesDoorShell": true,
+      "note": [
+        "Enter with G-mode direct, and quickly freeze the Metroids (if still alive) and Rinkas.",
+        "Climb up 1 screen, moving quickly to avoid Rinka damage.",
+        "Perform a turnaround buffered spin-jump away from the door to trigger the transition, bypassing any lock on the door."
+      ]
+    },
+    {
+      "link": [2, 1],
+      "name": "Very Deep Stuck X-Ray Climb With Crystal Flash",
+      "entranceCondition": {
+        "comeInWithGMode": {
+          "mode": "direct",
+          "morphed": false
+        }
+      },
+      "requires": [
+        {"or": [
+          "canArtificialMorph",
```

Comment:
artificialMorph doesn't say kill the metroids, or artificialMorphCrystalFlash, so you don't need it.  Entering with knockback gives IFrames for walking through the metroids.

Reply:
Makes sense. I was testing with my "doorhurt" patch which apparently doesn't give i-frames, so I didn't notice. Now that we have door portals in the practice hack, I should probably be using that instead.